installscripts/jazz-terraform-unix-noinstances/cognito.tf

```diff
@@ -1,15 +1,59 @@
 #
 # Cognito resources
 #
-resource "null_resource" "cognito_user_pool" {
-
-provisioner "local-exec" {
-    command = "${var.cognito_cmd}  ${var.envPrefix} ${var.envPrefix}-api-onboarding ${var.cognito_pool_username} ${var.cognito_pool_password} ${var.jenkinsjsonpropsfile}"
-  }  
-provisioner "local-exec" {
-    when = "destroy"
-  command = "${var.cognitoDelete_cmd}  ${var.envPrefix}"
+resource "aws_cognito_user_pool" "pool"{
+  name = "${var.envPrefix}"
+  username_attributes = ["email"]
+  schema = [
+    {
+      name = "email"
+      attribute_data_type = "String"
+      required = true
+    },
+    {
+      name = "reg-code"
+      attribute_data_type = "String"
+      string_attribute_constraints = {
+        min_length = 1
+      },
+    }
+  ]
+  tags = {
+    Application = "Jazz"
+    JazzInstance = "${var.envPrefix}"
+  }
+  auto_verified_attributes = ["email"]
+  verification_message_template = {
+    email_subject_by_link = "Jazz Notification - Account Verification"
+    email_message_by_link = "Hello,\n<br><br>\nThanks for signing up!\n<br><br>\nPlease click the link to verify your email address: {##VERIFY EMAIL##}\n<br><br>\nTo know more about Jazz, please refer to link https://github.com/tmobile/jazz-core/wiki\n<br><br>\nBest,<br>\nJazz Team"
+    default_email_option = "CONFIRM_WITH_LINK"
+  }
+  password_policy = {
+    minimum_length    = 6
```

Comment:
Update this to 8 (Gitlab requires 8 characters and Jazz UI enforces it)

Reply:
Done. It might be a good idea to strengthen the requirements for symbols and mixed case as well, but I was just trying to match what was already defined.